Objective-C/CBLReplicator.mm

```diff
@@ -193,7 +193,7 @@ - (void) resetRetryCount {
 - (void) _retry {
     CBL_LOCK(self) {
         CBLLogInfo(Sync, @"%@: Retrying...", self);
-        if (_repl || _state <= kCBLStateStopping) {
+        if ((_repl && (_state != kCBLStateOffline || _state != kCBLStateSuspended)) || _state <= kCBLStateStopping) {
```

Comment:
Hope we could simplify this logic but not the big deal. 

Another idea to make it readable is to do:
```
- (BOOL) isOffline { return (_state != kCBLStateOffline || _state != kCBLStateSuspended); }

if ((_repl && [self isOffline]) || _state <= kCBLStateStopping)
```
OR 
define IS_OFFLINE (_state != kCBLStateOffline || _state != kCBLStateSuspended)

Reply:
when c4replicator is lazily initialized, its state will be stopped. this means, 
1. if replicator goes offline. 
2. user check for pending-document-ids 
3. comes to online, and retries, it won't allow to start again(since _state = stopped). unless we mark `_state` to something else.

Checking for `_repl && ![self isOffline]` condition will always be false, since we always clear the replicator just before setting both `kCBLStateOffline` and `kCBLStateSuspended` state. 

am i right or missing something?

Reply:
I will verify this using a unit test and see.

Reply:
nevermind. i am wrong

Reply:
there is something wrong with this. I will investigate further